packages/nextjs/src/server/protect.ts

```diff
@@ -40,34 +59,34 @@ export const createProtect = (authObj: AuthObject): AuthProtect => {
     /**
      * User is not authenticated
      */
-    if (!authObj.userId) {
-      return handleUnauthorized();
+    if (!authObject.userId) {
+      return handleUnauthenticated ? handleUnauthenticated() : handleUnauthorized();
```

Comment:
We should probably opt out of the redirect to sign-in if the request refers to an endpoint and not a apge

Reply:
That is correct. Context for future reviewers: we're currently discussing this offline, will post more details soon.

Reply:
https://github.com/clerk/javascript/pull/2404/commits/0939fdab64bac1a159073c8a38fc1b5c824cef08